src/vulkan/device.cc

```diff
@@ -14,16 +14,62 @@
 
 #include "src/vulkan/device.h"
 
+#include <cassert>
+#include <cstring>
 #include <memory>
 #include <set>
 #include <vector>
 
 #include "src/make_unique.h"
+#include "src/vulkan/log.h"
 
 namespace amber {
 namespace vulkan {
 namespace {
 
+const char* const kRequiredValidationLayers[] = {
+#ifdef __ANDROID__
+    // Note that the order of enabled layers is important. It is
+    // based on Android NDK Vulkan document.
+    "VK_LAYER_GOOGLE_threading",      "VK_LAYER_LUNARG_parameter_validation",
+    "VK_LAYER_LUNARG_object_tracker", "VK_LAYER_LUNARG_core_validation",
+    "VK_LAYER_GOOGLE_unique_objects",
+#else   // __ANDROID__
+    // TODO(jaebaek): Handle cases running with other validation
+    //                layers e.g., MoltenVK.
+    "VK_LAYER_LUNARG_standard_validation",
+#endif  // __ANDROID__
+};
+
+const size_t kNumberOfRequiredValidationLayers =
+    sizeof(kRequiredValidationLayers) / sizeof(const char*);
+
+const char* kExtensionForValidationLayer = "VK_EXT_debug_report";
+
+VKAPI_ATTR VkBool32 VKAPI_CALL debugCallback(VkDebugReportFlagsEXT flag,
+                                             VkDebugReportObjectTypeEXT,
+                                             uint64_t,
+                                             size_t,
+                                             int32_t,
+                                             const char* layerPrefix,
+                                             const char* msg,
+                                             void*) {
+  const char* flag_message = nullptr;
+  switch (flag) {
+    case VK_DEBUG_REPORT_ERROR_BIT_EXT:
+      flag_message = "Error";
+      break;
+    case VK_DEBUG_REPORT_WARNING_BIT_EXT:
+      flag_message = "Warning";
+      break;
+    default:
+      flag_message = "No flag";
+      break;
+  }
+  LOGE("[%s] validation layer (%s): %s", flag_message, layerPrefix, msg);
```

Comment:
Let's not use a macro here. Make a method, pass in the string error message and call __android_log_write instead.